### PR TITLE
Stop using nightly images

### PR DESCRIPTION
**Pull request by miguelsorianod**
This PR performs the following changes:
* Stop referencing `nightly` tags for the 3scale components container images (system, zync, ...). Now `latest` tags are used.
* Stop building nightly images in the nightly job. It is important to notice that although it removes the build of nightly images it still runs some tests nightly
* Rename the CSV displayName to `3scale (development latest)`
* Rename the devnightly sample to devlatest
* Update templates
* Update code-level defaults to use `latest`
* Remove `pkg/3scale/amp/manual-templates/amp/build.yml`

### CODE REVIEW

**Comment by miguelsorianod** [comment]
ready @eguzki

File: .circleci/config.yml [hunk 1]
```diff
@@ -598,11 +598,7 @@ workflows:
             - generator
   nightly:
     jobs:
-      - build-push-3scale-images:
```
Comment (eguzki) [inline .circleci/config.yml:601]: `build-push-3scale-images` is being used? It should be removed if not used

**Review by eguzki — CHANGES_REQUESTED** [review]
rebase the branch to the current master

**Comment by eguzki** [comment]
`images-build` workflow should also be removed.

**Comment by eguzki** [comment]
`pkg/3scale/amp/manual-templates/amp/build.yml` is being used?

**Comment by miguelsorianod** [comment]
Done

**Review by eguzki — APPROVED** [review]

**Pull request merged**